__tests__/core/validation/validators/UrlValidator.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { UrlValidator } from '@/core/validation/validators'
+
+describe('UrlValidator.validate should', () => {
+  describe('return false when passing in', () => {
+    test('http://', () => {
+      expect(UrlValidator.validate('http')).toBeFalsy()
+    })
+    test('https://', () => {
+      expect(UrlValidator.validate('http://')).toBeFalsy()
+    })
+    test('https://foo', () => {
+      expect(UrlValidator.validate('https://q')).toBeFalsy()
+    })
+    test('https://foo..', () => {
+      expect(UrlValidator.validate('https://foo..')).toBeFalsy()
+    })
+    test('https://foo.-', () => {
+      expect(UrlValidator.validate('https://foo.-')).toBeFalsy()
+    })
+    test('https://foo--', () => {
+      expect(UrlValidator.validate('https://foo--')).toBeFalsy()
+    })
+    test('https://192.0.0', () => {
+      expect(UrlValidator.validate('https://192.0.0')).toBeFalsy()
+    })
+  })
+  describe('return true when passing in', () => {
+    test('google.com', () => {
+      expect(UrlValidator.validate('google.com')).toBeTruthy()
+    })
+    test('http://google.com', () => {
+      expect(UrlValidator.validate('http://google.com')).toBeTruthy()
+    })
+    test('https://google.com', () => {
+      expect(UrlValidator.validate('https://google.com')).toBeTruthy()
+    })
+    test('https://goo-gle.com', () => {
```

Comment:
test('https://goo_gle.com', () => {
      expect(UrlValidator.validate('https://goo_gle.com')).toBeTruthy()
    })

please add this test and fix the validation, seems the validator doesn't allow for underscores